lib/uuid.ex

```diff
@@ -450,6 +451,86 @@ defmodule UUID do
     "Invalid argument; Expected: :dns|:url|:oid|:x500|:nil OR String, String"
   end
 
+  @doc """
+  Generate a new UUID v6. This version uses a combination of one or more of:
+  unix epoch, random bytes, pid hash, and hardware address.
+
+  Accepts a `node_type` argument that can be either `:mac_address` or
+  `:random_bytes`. Defaults to `:mac_address`. However, if there is a security
+  concern with using a MAC address, use `:random_bytes`.
+
+  See the [RFC draft, section 3.3](https://tools.ietf.org/html/draft-peabody-dispatch-new-uuid-format-00#section-3.3)
+  for more information on the node parts.
+
+  ## Examples
+
+  ```elixir
+  iex> UUID.uuid6()
+  "1eb0cff9-b540-6a26-b8e5-3ccd365d5db4"
+
+  iex> UUID.uuid6(:default)
+  "1eb0cff9-ecd5-6888-a104-3ccd365d5db4"
+
+  iex> UUID.uuid6(:hex)
+  "1eb0cffaca7b607ab8573ccd365d5db4"
+
+  iex> UUID.uuid6(:urn)
+  "urn:uuid:1eb0cffb-440b-641c-9c32-3ccd365d5db4"
+
+  iex> UUID.uuid6(:raw)
+  <<30, 176, 207, 251, 201, 24, 108, 154, 142, 208, 60, 205, 54, 93, 93, 180>>
+
+  iex> UUID.uuid6(:slug)
+  "HrDP_FneYrafizzNNl1dtA"
+  ```
+
+  """
+  def uuid6(node_type \\ :mac_address, format \\ :default)
+      when node_type in [:mac_address, :random_bytes] do
+    uuid6(uuid1_clockseq(), uuid6_node(node_type), format)
+  end
+
+  @doc """
+  Generate a new UUID v6, with an existing clock sequence and node address. This
+  version uses a combination of one or more of: unix epoch, random bytes,
+  pid hash, and hardware address.
+
+  ## Examples
+
+  ```elixir
+  iex> UUID.uuid6()
+  "cdfdaf44-ee35-11e3-846b-14109ff1a304"
+
+  iex> UUID.uuid6(:default)
+  "cdfdaf44-ee35-11e3-846b-14109ff1a304"
+
+  iex> UUID.uuid6(:hex)
+  "cdfdaf44ee3511e3846b14109ff1a304"
+
+  iex> UUID.uuid6(:urn)
+  "urn:uuid:cdfdaf44-ee35-11e3-846b-14109ff1a304"
+
+  iex> UUID.uuid6(:raw)
+  <<205, 253, 175, 68, 238, 53, 17, 227, 132, 107, 20, 16, 159, 241, 163, 4>>
+
+  iex> UUID.uuid6(:slug)
+  "zf2vRO41EeOEaxQQn_GjBA"
+  ```
+
+  """
```

Comment:
None of these code examples correspond to this uuid6/3 so my suggestion would be to not include them and instead link to uuid6/0. (FWIW other functions have the same issue.)

```suggestion
  @doc """
  Generate a new UUID v6, with an existing clock sequence and node address.
  
  See `uuid6/2` for more information.
  """
```

Other than that, 👍 and one could wish it would have been part of the original RFC :)

Reply:
Good point, thanks. And yes, I wish this was part of the original RFC too!